lib/models/device.coffee

```diff
@@ -770,7 +766,7 @@ getDeviceModel = (deps, opts) ->
 		.then ({ application, device }) ->
 
 			if device.device_type isnt application.device_type
-				throw new Error("Incompatible application: #{applicationNameOrId}")
+				throw new errors.ResinInvalidDeviceType("Incompatible application: #{applicationNameOrId}")
```

Comment:
Should this be backported?

Reply:
It's a bit debatable. I think there are a few weird cases where this _could_ break your code (if you're catching ResinInvalidDeviceType specifically for some other reason, expecting `move` to fail differently). That's a bit tenuous, but there's no desperate need to backport things like this, so I'd rather bundle it into the big clear break just to be safe.